src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -411,15 +424,71 @@ public static AssemblyResultSet assembleReads(final AssemblyRegion region,
         }
     }
 
+    /**
+     * Helper method that handles the actual "GGA-like" Merging of haplotype alleles into an assembly result set.
+     *
+     * First this method will filter out haplotypes that contain alleles that have failed the pileup calling filtering steps,
+     * Then the list will attempt to poke into the haplotype list artificial haplotypes that have the found alleles present.
+     *
+     * @param region
+     * @param argumentCollection
+     * @param aligner
+     * @param refHaplotype
+     * @param assemblyResultSet
+     * @param pileupAllelesFoundShouldFilter
+     * @param pileupAllelesPassingFilters
+     * @return
+     */
+    public static AssemblyResultSet applyPileupEventsAsForcedAlleles(final AssemblyRegion region, final AssemblyBasedCallerArgumentCollection argumentCollection,
+                                                                      final SmithWatermanAligner aligner, final Haplotype refHaplotype,
+                                                                      final AssemblyResultSet assemblyResultSet, final List<VariantContext> pileupAllelesFoundShouldFilter,
+                                                                      final List<VariantContext> pileupAllelesPassingFilters, final boolean debug) {
+        List<Haplotype> haplotypesWithFilterAlleles = new ArrayList<>();
+        // IF we find pileup alleles that we want to filter... AND we are not generating the partially determined haplotypes,
+        // we resort to a messy approach where we filter alleles by throwing away every haplotype supporting an allele. This is
+        // very dangerous since this could easily destroy phased variants with the haplotype.
+        if (!pileupAllelesFoundShouldFilter.isEmpty() && !argumentCollection.pileupDetectionArgs.generatePDHaplotypes) {
+            // TODO this is a bad algorithm for bad people
+            for(VariantContext delVariant : pileupAllelesFoundShouldFilter) {
+                for (Haplotype hap : assemblyResultSet.getHaplotypeList()) {
+                    if (hap.getEventMap()==null) {
+                        if (!hap.isReference()) {
+                            //throw new RuntimeException("empty event map for haplotype" + hap);
```

Comment:
Yes... it turns out there were some edge cases in the haplotype construction code that caused this to throw exceptions in enough cases that i'm just going to ignore and let these through... I think a few of them at least were cases where the SmithWaterman failed (or our heuristics to prevent equivalent events were not enough to capture everything).